test/custom-attributes-test.js

```diff
@@ -0,0 +1,273 @@
+import { expect } from "chai";
+import { Wampy } from "../src/wampy.js";
+import WebSocket from "./fake-ws-custom-attrs.js";
+
+describe("Wampy.js Custom Attributes (WAMP spec 3.1)", function () {
+    this.timeout(1000);
+
+    let wampy;
+
+    before(function () {
+        globalThis.WebSocket = WebSocket;
+    });
+
+    beforeEach(function () {
+        wampy = new Wampy("ws://fake.server.url/ws/", {
+            realm: "AppRealm",
+            ws: WebSocket,
+        });
+    });
+
+    afterEach(async function () {
+        if (wampy.getOpStatus().code === 0) {
+            await wampy.disconnect();
+        }
+    });
+
+    describe("Client-to-Router Messages (Options/Details)", function () {
+        // WAMP SPEC: [HELLO, Realm|string, Details|dict]
+        it("HELLO.Details - supports custom attributes via helloCustomDetails", async function () {
+            const helloCustomDetails = {
+                _client_version: "7.2.0",
+                _tracking_session: "session_12345",
+                _environment: "production",
+            };
+
+            const wampyWithCustomHello = new Wampy("ws://fake.server.url/ws/", {
+                realm: "AppRealm",
+                ws: WebSocket,
+                helloCustomDetails,
+            });
+
+            // This should work - HELLO uses helloCustomDetails, not _extractCustomOptions
+            await wampyWithCustomHello.connect();
+            expect(wampyWithCustomHello.getOpStatus().code).to.equal(0);
+            await wampyWithCustomHello.disconnect();
+        });
+
+        // WAMP SPEC: [SUBSCRIBE, Request|id, Options|dict, Topic|uri]
+        it("SUBSCRIBE.Options - supports custom attributes", async function () {
+            await wampy.connect();
+
+            const customOptions = {
```

Comment:
Okay. Thnx for removing some AI-garbage.... 
But I still don't see how `customOptions` are tested....
If they are not passed or are not correctly processed - then there will be no data-matching and default response for msg type will be returned. That's it. Where does the actual test happen?